rotkehlchen/chain/ethereum/interfaces/balances.py

```diff
@@ -34,19 +41,100 @@ def __init__(
+            chain: 'SupportedBlockchain',
             counterparty: PROTOCOLS_WITH_BALANCES,
     ):
         self.counterparty = counterparty
         self.event_db = DBHistoryEvents(database)
         self.evm_inquirer = evm_inquirer
+        self.chain = chain
+
+    @abc.abstractmethod
+    def get_gauge_deposit_events(self) -> set[tuple[HistoryEventType, HistoryEventSubType]]:
+        """
+        Common method for all the classes implementing this interface. It returns the gauge
+        deposit events. Staking to a gauge is done with a gauge deposit transaction which
+        is decoded to a set of events. One of these events represents the deposit action.
+        This function should return the type and subtype of the event that represents this
+        deposit action.
+        """
 
     @abc.abstractmethod
-    def query_balances(self) -> dict[ChecksumEvmAddress, dict['EvmToken', 'Balance']]:
+    def get_gauge_address(self, event: 'EvmEvent') -> Optional[ChecksumEvmAddress]:
+        """
+        Common method for all the classes implementing this interface. It returns the gauge
+        address from the event that represents the gauge deposit action or None.
         """
-        Common method for all the classes implementing this interface that returns a mapping
-        of user addresses to the their token balances. This is later called in
-        `query_eth_balances`.
+
+    def _query_gauges_balances(
+            self,
+            user_address: ChecksumEvmAddress,
+            gauges_to_token: dict[ChecksumEvmAddress, EvmToken],
+            call_order: list[WeightedNode],
+            chunk_size: int,
+            balances_contract: Callable,
+    ) -> dict[EvmToken, Balance]:
+        """
+        Query the set of gauges in gauges_to_token and return the balances for each
+        lp token deposited in all gauges.
         """
+        balances: dict[EvmToken, Balance] = defaultdict(Balance)
+        gauge_chunks = get_chunks(list(gauges_to_token.keys()), n=chunk_size)
+        for gauge_chunk in gauge_chunks:
+            tokens = [gauges_to_token[staking_addr] for staking_addr in gauge_chunk]
+            gauges_balances = balances_contract(
+                address=user_address,
+                staking_addresses=gauge_chunk,
+                tokens=tokens,
+                call_order=call_order,
+            )
+
+            # Now map the gauge to the underlying token  todo is this the same for velodrome?
+            for lp_token, balance in gauges_balances.items():
+                lp_token_price = Inquirer().find_usd_price(lp_token)
+                balances[lp_token] += Balance(
+                    amount=balance,
+                    usd_value=lp_token_price * balance,
+                )
+
+        return balances
+
+    def query_balances(self) -> BalancesType:
+        """
+        Query gauge balances for the addresses that have interacted with known gauges.
+        """
+        balances: BalancesType = {}
```

Comment:
As it is now the query balacnes method can be used only for gauge like products. I would let it as an abstract method like before and create a subclass of this interface, let's say `ProtocolWithGauges` and then implement the logic here